kernel/src/main/java/org/cristalise/kernel/lifecycle/instance/Activity.java

```diff
@@ -441,7 +441,12 @@ private boolean checkParentFinishable(CompositeActivity parent, Vertex[] outVert
             hasNoNext = parent.getPossibleActs(outVertex, GraphTraversal.kUp).size() == 0;
         }
         else if (outVertex instanceof Loop) {
-            hasNoNext = parent.getPossibleActs(outVertex, GraphTraversal.kDown).size() == 0;
+            //Check upper AndSplit first
+            hasNoNext = parent.getPossibleActs(outVertex, GraphTraversal.kUp).size() == 0;
```

Comment:
There is a better way to check if the Loop has activity enabled in its body.

Reply:
Modification done. The tests of testing project have been runed and passed.